examples/ErrorHandling/HandleRateExceededError.cs

```diff
@@ -0,0 +1,192 @@
+                t.Start();
+                tasks.Add(t);
+            }
+
+            Task.WaitAll(tasks.ToArray());
+        }
+
+        /// <summary>
+        /// Displays the result from the mutate operation.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The Google Ads customer ID for which the call is made.</param>
+        /// <param name="adGroupId">The ad group to which keywords are added.</param>
+        /// <param name="threadIndex">The thread ID.</param>
+        private async Task CreateKeyword(GoogleAdsClient client, int threadIndex, long customerId,
+            long adGroupId)
+        {
+            // Get the AdGroupCriterionServiceClient.
+            AdGroupCriterionServiceClient adGroupCriterionService =
+                client.GetService(Services.V2.AdGroupCriterionService);
+
+            List<AdGroupCriterionOperation> operations = new List<AdGroupCriterionOperation>();
+
+            for (int i = 0; i < NUM_KEYWORDS; i++)
+            {
+                AdGroupCriterion criterion = new AdGroupCriterion()
+                {
+                    Keyword = new KeywordInfo()
+                    {
+                        Text = $"mars cruise thread {threadIndex} seed {i}",
+                        MatchType = KeywordMatchType.Exact
+                    },
+                    AdGroup = ResourceNames.AdGroup(customerId, adGroupId),
+                    Status = AdGroupCriterionStatus.Paused
+                };
+
+                // Creates the operation.
+                operations.Add(new AdGroupCriterionOperation() { Create = criterion });
+            }
+
+            int retryCount = 0;
+            int retrySeconds = 10;
+            const int NUM_RETRIES = 3;
+
+            while (retryCount < NUM_RETRIES)
+            {
+                try
+                {
+                    // Makes the validateOnly mutate request.
+                    MutateAdGroupCriteriaResponse response =
+                        await adGroupCriterionService.MutateAdGroupCriteriaAsync(
+                            customerId.ToString(), operations, false, true);
+                    Console.WriteLine($"Validated {operations.Count} ad group criteria:");
+                    break;
+                }
+                catch (GoogleAdsException e)
+                {
+                    // Checks if any of the errors are QuotaError.RESOURCE_EXHAUSTED or
+                    // QuotaError.RESOURCE_TEMPORARILY_EXHAUSTED.
+                    if (e.Failure.Errors.Where(err =>
+                        err.ErrorCode.QuotaError == QuotaError.ResourceExhausted ||
+                        err.ErrorCode.QuotaError == QuotaError.ResourceTemporarilyExhausted)
+                        .Count() > 0)
+                    {
+                        Console.Error.WriteLine(
+                            $"Received rate exceeded error, retry after {retrySeconds} seconds.");
```

Comment:
Should we also demonstrate how to use the Scope of the error like we did in https://developers.google.com/adwords/api/docs/samples/csharp/error-handling#handle-rate-exceeded-errors?